### DISCUSSION TITLE
What is the anonymous author identifier based on?

### DISCUSSION THREAD

**Discussion by greyidol** [Retrospring/retrospring#1057] [Q&A] [answered]
I currently have my nginx set to anonymize IPs for privacy (sets them to 127.0.0.2 except for Tor, which is 127.0.0.1). Is this why all my anon questions have the same author identifier?

**Comment by raccube** [comment, marked as answer]
The anonymous author identifier is a hashed value based off of the request IP. 
The code responsible for this can be found here:
https://github.com/Retrospring/retrospring/blob/main/app/models/anonymous_block.rb#L10
https://github.com/Retrospring/retrospring/blob/main/app/controllers/ajax/question_controller.rb#L34

If you want this data to be accessible by the application when proxied via nginx, you will need to set a `X-Forwarded-For` or `X-Real-IP` header.

**Reply by greyidol** [reply]
Is there any way to get the anonymous identifier to work while also not logging IPs or showing the user's IP in settings?

**Reply by greyidol** [reply]
I figured it out. I set a proper X-Forwarded-For and X-Real-IP header, but I also use the following database trigger:

```sql
CREATE FUNCTION delete_ip() RETURNS trigger AS $$
BEGIN
	NEW.current_sign_in_ip = CASE WHEN NEW.current_sign_in_ip = '127.0.0.1' THEN NEW.current_sign_in_ip ELSE '127.0.0.2' END;
	NEW.last_sign_in_ip = CASE WHEN NEW.last_sign_in_ip = '127.0.0.1' THEN NEW.last_sign_in_ip ELSE '127.0.0.2' END;
	RETURN NEW;
END;
$$ LANGUAGE plpgsql;

CREATE TRIGGER delete_ips                                                    
BEFORE INSERT OR UPDATE ON users
FOR EACH ROW EXECUTE PROCEDURE delete_ip();
```

That way, the anonymous identifiers still work (at least for non-onion users) but the IPs are not logged in user settings.
